### PR TITLE
FIX include 'n' field in 1D/2D list + remove singleton dimension in 1D dict

### LINKED ISSUE #875: Why wouldn't returned dicts with `out_type="list"` not have an `n` field?
Astutely brought up by reviewer @MuawizChaudhary during CR of #862 
https://github.com/kymatio/kymatio/pull/862#discussion_r895089212

In 1D, the `scattering` method ends on a rather strange note:
```python
...
elif out_type == 'list':
    for x in out_S:
        x.pop('n')
return out_S
```

in other words, we're going out of our way to conceal the `n` field to the `out_type="list"` user.
Personally, i don't see a justification for not exposing `n`. It is quite easy to understand and re-use: it can be seen as a "pointer" (or tuple of pointers) to the filterbank `psi1_f` (or tuple of filterbanks `(psi1_f, psi2_f)`)
And `n` is already exposed to the user anyway, under the form of `S.keys()` with `out_type="dict"` (#861). So i don't think we're running into a big risk of introducing a new functionality that will be difficult to maintain in the future.

Hence i propose to experiment with removing this `x.pop('n')` statement. And if the tests pass, make it part of `v0.3`. There's nothing in it that couldn't be achieved with `out_dict="dict"` and functional programming / some advanced looping mechanisms (like `for k,v in S.items()`, but not all our users are familiar with those Pythonic idioms, so giving them access to `n` as the value of a simple data type (dict) contained in a list is a good way to let them write custom code that inspects `psi` metadata n relation with `S` activations.

This change would remove code in 1D and add very little code in 2D/3D.

Thoughts?

### LINKED ISSUE #895: what should be the returned shape of `Scattering1D` paths under `out_type='dict'` ?
Spotted while reviewing #871 with @janden

Right now, `out_type='dict'`behaves strangely. 

```python
J, Q = 2, 1
length = 1024
S = Scattering1D(J, shape, Q)
x = torch.zeros((5, 10,) + shape)

Sx = S(x)
print(Sx.shape)

S.out_type = "dict"
Sx = S(x)
print(list(map(lambda v: v.shape, Sx.values())))
```

Result: 
```
torch.Size([5, 10, 4, 256])

[torch.Size([5, 10, 1, 256]),
 torch.Size([5, 10, 1, 256]),
 torch.Size([5, 10, 1, 256]),
 torch.Size([5, 10, 1, 256])]
```

In other words, `out_type="dict"` is inserting a singleton dimension in penultimate position. This penultimate position serves to concatenate coefficients under `out_type="array"` (it becomes `4` in the example above).

I am in favor of removing this extra singleton dimension, and making values in `out_type="dict"` more like `'coef'`'s of `out_type="list"`. Compare with 

```python
S.out_type = "list"
Sx = S(x)
list(map(lambda v: v['coef'].shape, Sx))
```

Result:

```
[torch.Size([5, 10, 256]),
 torch.Size([5, 10, 256]),
 torch.Size([5, 10, 256]),
 torch.Size([5, 10, 256])]
```

This would be a breaking change, which is why i didn't make it single-handedly in my open PR #871.
I'm curious to hear everyone's thought here? bug or feature?

### PR DESCRIPTION
**Pull request by lostanlen**
Fix #875, Fix #895
as discussed in today's Kymatio meeting

NB: these changes must be in the release notes.

### CODE REVIEW

File: kymatio/scattering1d/frontend/numpy_frontend.py [hunk 1]
```diff
@@ -39,12 +39,11 @@ def scattering(self, x):
             for k, v in S.items():
                 # NOTE: Have to get the shape for each one since we may have
                 # average == False.
-                scattering_shape = v.shape[-2:]
```
Comment (MuawizChaudhary) [inline kymatio/scattering1d/frontend/numpy_frontend.py:42]: why this change?

Comment (MuawizChaudhary) [reply kymatio/scattering1d/frontend/numpy_frontend.py:42]: Does it remove the 1 dimension?

**Review by MuawizChaudhary — APPROVED** [review]

File: kymatio/scattering1d/frontend/numpy_frontend.py (hunk 1 above)
Comment (lostanlen) [reply kymatio/scattering1d/frontend/numpy_frontend.py:42]: yes

**Comment by MuawizChaudhary** [comment]
@changhongw If you approve pls press the big green button!

**Comment by changhongw** [comment]
Thanks for noticing this problem!
